flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSource.java

```diff
@@ -208,21 +218,30 @@ public FlinkInputFormat buildFormat() {
 
       if (projectedSchema == null) {
         contextBuilder.project(icebergSchema);
+      } else if (projectedFields != null) {
+        // Push down the nested projection so that don't need to get extra fields when get data from files.
+        Schema icebergProjectionSchema = projectSchema(icebergSchema);
+        contextBuilder.project(icebergProjectionSchema);
+        projectedFields = mappingNestProjectedFields(icebergProjectionSchema);
```

Comment:
Get the mapping between the underlying iceberg nested schema and the Flink source output schema.